File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} k
     * @return {number}
     */
    kthSmallest(root, k) {

        if(root===null) return null
        const res=[]

        function inOrder(node=root)
        {
            if(node===null) return 
            inOrder(node.left)
            res.push(node.val)
            inOrder(node.right)
        }
        inOrder()

        return res[k-1]
    }
}

File: Data Structures & Algorithms/sort-an-array/submission-0.go
func sortArray(nums []int) []int {
    if len(nums)<= 1 {
		return nums
	}
	left:=sortArray(nums[:len(nums)/2])
	right:=sortArray(nums[len(nums)/2:])

	return merge(left,right)
}

func merge(left, right []int) []int {
	res:=make([]int,0,len(left)+len(right))
	i,j:=0,0

	for i<len(left) && j <len(right) {

		if left[i]<right[j] {
			res=append(res,left[i])
			i++
		} else {
			res=append(res,right[j])
			j++
		}
	}
	res=append(res,left[i:]...)
	res=append(res,right[j:]...)
	return res
}